lib/Dialect/FIRRTL/Transforms/RemoveUnusedPorts.cpp

```diff
@@ -79,12 +79,43 @@ void RemoveUnusedPortsPass::removeUnusedModulePorts(
     if (port.isInput() && !arg.use_empty())
       continue;
 
+    auto portIsUnused = [&](InstanceRecord *a) -> bool {
+      auto port = a->getInstance()->getResult(arg.getArgNumber());
+      return port.getUses().empty();
+    };
+
     // Output port.
     if (port.isOutput()) {
       if (arg.use_empty()) {
         // Sometimes the connection is already removed possibly by IMCP.
         // In that case, regard the port value as an invalid value.
         outputPortConstants.push_back(None);
+      } else if (llvm::all_of(instanceGraphNode->uses(), portIsUnused)) {
+        // If the port is unused in all of its instantiations, then replace
+        // internal references to the port with whatever is driving it.
+        StrictConnectOp connect;
+        for (auto &use : arg.getUses())
+          if (auto maybeConnect = dyn_cast<StrictConnectOp>(use.getOwner()))
+            if (maybeConnect.dest() == arg)
+              connect = maybeConnect;
+        assert(connect && "unable to find sole driver of port");
```

Comment:
Could you just abort when single connect is not found rather causing an assertion failure?
It might cause an assertion failure in aggregate preservation mode if ports are connected with subfield.

Reply:
This is now fixed. A wire is unconditionally created, regardless of the number of drivers. That then means you wind up with a multiply driven wire. This should be impossible to reach, but also preserves the original semantic without having to special case anything.